applications/crossbar/src/modules/cb_callflows.erl

```diff
@@ -187,9 +187,12 @@ request_patterns(Context) ->
 validate_request(CallflowId, Context) ->
     case request_numbers(Context) of
         [] -> validate_patterns(CallflowId, Context);
-        OriginalNumbers
-          when is_list(OriginalNumbers) ->
-            validate_callflow_schema(CallflowId, normalize_numbers(Context, OriginalNumbers));
+        OriginalNumbers when is_list(OriginalNumbers) ->
+            Validators = [{fun validate_min_length_numbers/3, [OriginalNumbers]}
```

Comment:
At the moment, I feel like you could just edit the schema with the system_config-defined length and then validate as before. Then you don't need this custom validator code to check length, right?

Reply:
For this change, the default minimum length is defined in system_config, but can be overridden at an account level. Relying on the system_config schema alone for validation will not allow the minimum length to be overridden for a particular account, which is what the kapps_account_config:get_global() call within the new validator code gives us. 

Also we don't want to enforce minimum length for existing callflow numbers. So the unchanged check within the validator code is needed to ensure the user can update and save a callflow without being blocked by validation of an existing number that does not meet minimum length. 

Does that make sense? Is there an easier way of doing this, using validate_callflow_schema, that I've missed?

Reply:
My initial preference is to put this behind a config toggle; if an account opts in then all callflow changes will be subject to it.

My pseudocode would then be:

```erlang
    OriginalNumbers ->
        validate_numbers(CallflowId, Context, OriginalNumbers, kapps_config:is_true(?APP_NAME, <<"enforce_minimum_length">>, 'false'))
end.

validate_numbers(CallflowId, Context, OriginalNumbers, 'false') ->
    validate_callflow_schema/2...;
validate_numbers(CallflowId, Context, OriginalNumbers, 'true') ->
    MinLength = kapps_account_config:get(),
    Schema = kz_json_schema:get(<<"callflows">>),
    Updated = add_min_length_check(Schema, MinLength),
    validate_callflow_schema/2...
```

So if the sysadmin opts into min-length checks, adjust the schema to use the account's configured min length and apply them to any callflow changes (new or edits).

Thoughts?

Reply:
Thanks for the suggestion, I can see your point. I'm going to try this approach and will get back to you with an updated solution

Reply:
Just to clarify, should the config toggle be at a system level or at an account level? I realise the example is just pseudocode but kapp_configs module accesses system_config, while it sounds like the decision to enforce minimum length should be at an account config level. Can you confirm your preference?

Reply:
@aisconn System-level config for whether to enforce the minimum or not; account-level config for the minimum itself.